drivers/net/ethernet/vertexcom/mse102x.c

```diff
@@ -308,27 +311,33 @@ static void mse102x_rx_pkt_spi(struct mse102x_net *mse)
 	__be16 rx = 0;
 	u16 cmd_resp;
 	u8 *rxpkt;
+	bool drop = false;
 
 	mse102x_tx_cmd_spi(mse, CMD_CTR);
 	if (mse102x_rx_cmd_spi(mse, (u8 *)&rx))
 		return;
 
 	cmd_resp = be16_to_cpu(rx);
-	if ((cmd_resp & CMD_MASK) != CMD_RTS) {
+	if ((cmd_resp & CMD_MASK) == CMD_RTS) {
+		rxlen = cmd_resp & LEN_MASK;
+		if (rxlen < ETH_ZLEN) {
```

Comment:
Now that we know that the rxlen should be max 1518, I'd add a check here too using the defines ETH_FRAME_LEN + VLAN_HLEN